cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py

```diff
@@ -85,36 +143,123 @@ def create_task(task_class, options=None, project_config=None, org_config=None):
     return create_task
 
 
-def pytest_configure(config):
-    # register an additional marker
-    config.addinivalue_line(
-        "markers",
-        "integration_test(): an integration test that should only be executed when requested",
-    )
-    config.addinivalue_line(
-        "markers",
-        "no_vcr(): an integration test that should not attempt to store VCR cassettes",
-    )
+def vcr_cassette_name_for_item(item):
+    """Name of the VCR cassette"""
+    test_class = item.cls
+    if test_class:
+        return "{}.{}".format(test_class.__name__, item.name)
+    return item.name
 
 
-def pytest_runtest_setup(item):
-    is_integration_test = any(item.iter_markers(name="integration_test"))
-    if is_integration_test:
-        if not item.config.getoption(
-            "--accelerate-integration-tests"
-        ) and not item.config.getoption("--org"):
-            pytest.skip("test requires --org or --accelerate-integration-tests")
-        no_vcr = any(item.iter_markers(name="no_vcr"))
-        if no_vcr and item.config.getoption("--accelerate-integration-tests"):
-            pytest.skip("test cannot be accelerated. " "It is not compatible with VCR.")
-
-
-@pytest.fixture(scope="module")
-def vcr_cassette_dir(request):
-    is_integration_test = any(request.node.iter_markers(name="integration_test"))
-    test_dir = request.node.fspath.dirname
-    if is_integration_test:
-        return os.path.join(test_dir, "large_cassettes")  # 8-tracks
-    else:  # standard behaviour from
-        # https://github.com/ktosiek/pytest-vcr/blob/master/pytest_vcr.py
-        return os.path.join(test_dir, "cassettes")
+def classify_and_modify_test(item, marker_names):
+    if "slow" in marker_names:
+        if not item.config.getoption("--run-slow-tests"):
+            pytest.skip("slow: test requires --run-slow-tests")
+
+    if "large_vcr" in marker_names:
+        library_dir = Path(cumulusci.__file__).parent.parent / "large_cassettes"
+        item.add_marker(pytest.mark.vcr(cassette_library_dir=str(library_dir)))
+
+        test_name = vcr_cassette_name_for_item(item)
+        test_path = Path(library_dir) / (test_name + ".yaml")
+        if not (test_path.exists() or item.config.getoption("--org")):
+            pytest.skip("large_vcr: test requires --org to generate large cassette")
+
+    elif "needs_org" in marker_names and not item.config.getoption("--org"):
+        pytest.skip("needs_org: test requires --org")
+
+
+# a cache of orgs matching shapes:
+#    TODO test that it is properly used
+org_shapes = ContextVar("org_shapes", default={})
+
+
+@pytest.fixture(
+    scope="function",
+)
+def run_code_without_recording(request, vcr, user_requested_network_access):
+    def really_run_code_without_recording(func):
+        if user_requested_network_access:
+            # Run the setup code, but don't record it
+            with set_mode(RECORDING, False), set_mode(READING, False):
+                return func()
```

Comment:
Silently doing nothing when there's no org present seems surprising. Where would this be used such that it could run in a test that isn't being skipped with no org?